rdflib/plugins/sparql/aggregates.py

```diff
@@ -98,10 +100,14 @@ def agg_Count(a, group, bindings):
 
 
 def agg_Sample(a, group, bindings):
-    try:
-        bindings[a.res] = _eval(a.vars, iter(group).next())
-    except StopIteration:
-        pass  # no res
+    for ctx in group:
+        val = _eval(a.vars, ctx)
+        if type(val) is not NotBoundError:
+            bindings[a.res] = val
+            break
+    else:
+        bindings[a.res] = Literal(0)
```

Comment:
yes, the problem is that there are various meanings of "error"... IMO they don't mean to remove the whole result row / cancel the whole query, but just the one binding within the row, as shown here:
http://www.w3.org/TR/sparql11-query/#aggregateExample2

So if we encounter such an error, i think it's correct to somehow return the row with a placeholder indicating an unbound variable (which i thought `None` was), rather than removing the result row or simply cancelling the whole query.

At least that's what Virtuoso does in the sample query above, so it must be true ;)